charts/redpanda/templates/_configmap.tpl

```diff
@@ -130,7 +130,7 @@ redpanda.yaml: |
 {{- end }}
 {{- with dig "node" dict .Values.config }}
   {{- range $key, $element := .}}
-    {{- if and (or (eq (typeOf $element) "bool") $element) (and (eq $key "crash_loop_limit") (include "redpanda-atleast-23-1-1" $root | fromJson).bool) }}
+    {{- if or (eq (typeOf $element) "bool") (and (eq $key "crash_loop_limit") (include "redpanda-atleast-23-1-1" $root | fromJson).bool) }}
```

Comment:
```suggestion
    {{- if and (or (eq (typeOf $element) "bool") $element) (or (and (eq $key "crash_loop_limit") (include "redpanda-atleast-23-1-1" $root | fromJson).bool) (ne $key "crash_loop_limit")) }}
```

The first check `or (eq (typeOf $element) "bool") $element` needs to stay as is. Bug was introduced in second part of the condition statement. The key crash_loop_limit can always be defined, but it needs to be pass only if Redpanda semver is at least 23.1.1. The rest of the keys should be pass as is.

Reply:
Sharp eye! This syntax remains hard to read. Fixed 👍

Reply:
Indeed 😮‍💨 I'm not fun of go template.